### ISSUE TITLE
RKE2 Cluster Provisioning fails when a cluster with the same name existed before

### ISSUE THREAD

**Issue by bashofmann** [rancher/rancher#33897]
**Rancher Server Setup**
- Rancher version: master-head#30574c2
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): K3s v1.21.3+k3s1
- Proxy/Cert Details: No proxy, valid Let's Encrypt certificate

**Information about the Cluster**
- Kubernetes version: v1.21.3-rc1+rke2r2
- Cluster Type (Local/Downstream): downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): infrastructure provider, tested DO and AWS

**Describe the bug**
RKE2 Cluster Provisioning fails when a cluster with the same name existed before.

**To Reproduce**
* Create a cluster at DigitalOcean with the name `dotest`
* Wait until the cluster has been created successfully
* Delete the cluster
* Create a new cluster at DigitalOcean with the same name `dotest`

**Result**
The second cluster fails with the error message

```
Error while applying agent YAML, it will be retried automatically: exit status 1, error: You must be logged in to the server (the server has asked for the client to provide credentials)
```

**Expected Result**
The second cluster is created successfully.

**Screenshots**
![Bildschirmfoto 2021-08-03 um 16 01 33](https://user-images.githubusercontent.com/243056/128029353-3f149100-a9f1-4762-8387-c82630d9f912.png)

**Comment by thedadams** [comment]
@deniseschannon This is a known issue and should be fixed in https://github.com/rancher/fleet/issues/478

**Comment by deniseschannon** [comment]
This should be available now to-test. @bashofmann Can you take a look at it? It woudl be available with the latest fleet update.

**Comment by davidnuzik** [comment]
We should link the PR to this issue.

**Comment by davidnuzik** [comment]
**My checks PASSED**
 
**Reproduction Steps:**

Not required as I fully understand this issue.
 
**Validation Steps:**

Rancher version: v2.6-head 4def65610 8/13/21 10:50 am Pacific
Rancher cluster type: single-node docker
Docker version: 19.03
 
Downstream cluster type: rke2
Downstream K8s version: v1.21.3-rc5+rke2r2
 
Validation steps:

1. Create an rke2 cluster, mine was named "dave-try-1node".
2. Wait until the cluster has been created successfully.
3. Delete the cluster.
4. Create the same cluster again with the same exact name.

Result:
Confirmed I did NOT get the error message mentioned in the GitHub issue body.
Confirmed the cluster came up successfully.
Confirmed I could deploy a workload without any issues.
